src/editor/overlay_widget.cpp

```diff
@@ -184,11 +232,26 @@ EditorOverlayWidget::draw_rectangle()
   for (int x = static_cast<int>(dr.get_left()); x <= static_cast<int>(dr.get_right()); x++, x_++) {
     int y_ = sgn_y ? 0 : static_cast<int>(-dr.get_height());
     for (int y = static_cast<int>(dr.get_top()); y <= static_cast<int>(dr.get_bottom()); y++, y_++) {
-      input_tile( Vector(static_cast<float>(x), static_cast<float>(y)), m_editor.get_tiles()->pos(x_, y_) );
+      if (autotile_mode) {
+        input_autotile( Vector(static_cast<float>(x), static_cast<float>(y)), m_editor.get_tiles()->pos(x_, y_) );
+      } else {
+        input_tile( Vector(static_cast<float>(x), static_cast<float>(y)), m_editor.get_tiles()->pos(x_, y_) );
+      }
     }
   }
 }
 
+bool
+EditorOverlayWidget::check_tiles_for_fill(uint32_t replace_tile, uint32_t target_tile, uint32_t third_tile)
+{
+  if (autotile_mode) {
+    return AutotileSet::get_tileset_from_tile(replace_tile) == AutotileSet::get_tileset_from_tile(target_tile)
```

Comment:
the line width hurts me a bit. Not sure if there's anything that can be done about that.